billing/views.py

```diff
@@ -217,26 +235,36 @@ def customer_subscription_created(self, subscription: stripe.Subscription) -> No
         self._log_updated(1)
 
     def customer_subscription_updated(self, subscription: stripe.Subscription) -> None:
-        owner: Owner = Owner.objects.get(
+        owners: QuerySet[Owner] = Owner.objects.filter(
             stripe_subscription_id=subscription.id,
             stripe_customer_id=subscription.customer,
         )
+        if not owners.exists():
+            log.error(
+                "Subscription update requested with invalid plan",
```

Comment:
I think this log message can be reworded; it's not necessarily due to invalid plan but because we couldn't find the owner for the combo